Kernel/Ordering.cpp

```diff
@@ -891,40 +901,42 @@ DArray<int> PrecedenceOrdering::predLevelsFromOptsAndPrec(Problem& prb, const Op
 
 void PrecedenceOrdering::show(ostream& out) const 
 {
-  CALL("PrecedenceOrdering::show(ostream& out)")
-  {
-    out << "% Function precedences, smallest symbols first (line format: `<name> <arity>`) " << std::endl;
-    out << "% ===== begin of function precedences ===== " << std::endl;
-    DArray<unsigned> functors;
+  CALL("PrecedenceOrdering::show(ostream& out)");
+
+  auto _show = [&](const char* precKind, unsigned cntFunctors, auto getSymbol, auto compareFunctors)
+    {
+      out << "% " << precKind << " precedences, smallest symbols first (line format: `<name> <arity>`) " << std::endl;
+      out << "% ===== begin of " << precKind << " precedences ===== " << std::endl;
+      DArray<unsigned> functors;
+
+      functors.initFromIterator(getRangeIterator(0u, cntFunctors), cntFunctors);
+      functors.sort(closureComparator(compareFunctors));
+      for (unsigned i = 0; i < cntFunctors; i++) {
+        auto sym = getSymbol(functors[i]);
+        out << "% " << sym->name() << " " << sym->arity() << std::endl;
+      }
 
-    functors.initFromIterator(getRangeIterator(0u,env.signature->functions()),env.signature->functions());
-    functors.sort(closureComparator([&](unsigned l, unsigned r){ return intoComparison(compareFunctionPrecedences(l,r)); }));
-    for (unsigned i = 0; i < functors.size(); i++) {
-      auto sym = env.signature->getFunction(functors[i]);
-      out << "% " << sym->name() << " " << sym->arity() << std::endl;
-    }
+      out << "% ===== end of " << precKind << " precedences ===== " << std::endl;
 
-    out << "% ===== end of function precedences ===== " << std::endl;
-  }
+      out << "%" << std::endl;
+    };
 
-  out << "%" << std::endl;
+  _show("type constructor", 
+      env.signature->typeCons(), 
+      [](unsigned f) { return env.signature->getTypeCon(f); },
+      [&](unsigned l, unsigned r){ return intoComparison(compareFunctionPrecedences(l,r)); });
```

Comment:
ah yeah .. is this the correct way of comparing type constructor precedences?
I copied it from your code, but I reckon that should be compareTypeConPrecedences, right?

Reply:
Thanks, fixed!